tests/test_assert_assume.cpp

```diff
@@ -0,0 +1,127 @@
+/* Copyright 2019-Barefoot Networks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <gtest/gtest.h>
+
+#include <bm/bm_sim/actions.h>
+#include <bm/bm_sim/core/primitives.h>
+#include <bm/bm_sim/phv.h>
+#include <bm/bm_sim/parser_error.h>
+#include <bm/bm_sim/packet.h>
+
+using namespace bm;
+
+class assert_ : public ActionPrimitive<const Data &> {
```

Comment:
if you define them in the core primitives, you won't need to duplicate the code

Reply:
Primitives in core/primitives.cpp do abort() when condition is false, so it made my test to fail. That is why I needed these primitives to mark that assert/assume was triggered.

Reply:
Did you try this: https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#how-to-write-a-death-test

Reply:
Thank you for the suggestion, I was not familiar with this. I will definitely  try this one.